server/sonar-server/src/main/java/org/sonar/server/computation/step/QualityGateMeasuresStep.java

```diff
@@ -245,4 +265,34 @@ public Condition apply(@Nonnull EvaluatedCondition input) {
       return input.getCondition();
     }
   }
+
+  private static class MetricEvaluationResult {
+    private final EvaluationResult evaluationResult;
+    private final Condition condition;
+
+    private MetricEvaluationResult(EvaluationResult evaluationResult, Condition condition) {
+      this.evaluationResult = evaluationResult;
+      this.condition = condition;
+    }
+  }
+
+  private enum ConditionToMetric implements Function<Condition, Metric> {
+    INSTANCE;
+
+    @Override
+    @Nonnull
+    public Metric apply(@Nonnull Condition input) {
+      return input.getMetric();
+    }
+  }
+
+  public enum ConditionToPeriod implements Function<Condition, Integer> {
+    INSTANCE;
+
+    @Override
+    @Nullable
+    public Integer apply(@Nonnull Condition input) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this "Nonnull" annotation to honor the overridden method's contract. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2638)